src/components/app/app.component.tsx

```diff
@@ -1,75 +1,57 @@
-import React, { useState, FC } from 'react';
-import Main from '@app/components/main/main.component';
-import MoviesService from '@app/services/movies.service';
-import Header from '@app/components/header/header.component';
-import Footer from '@app/components/footer/footer.component';
+import React, {
+  useState, FC, lazy, Suspense,
+} from 'react';
+import store from '@app/store/store';
+import { StoreContext } from 'storeon/react';
+import { IMovie } from '@shared/interfaces/movies.model';
 import useStyle from '@app/components/app/app.component.styles';
 import CoreModal from '@app/components/modals/coreModal/coreModal.component';
-import { Genres, IMovie, IMovieSortOptions } from '@app/mockData/movies.model';
 import MovieDetails from '@app/components/movieDetails/movieDetails.component';
 import HeaderActiveComponent from '@app/components/app/app.component.interface';
 import ErrorBoundary from '@app/components/errorBoundary/errorBoundary.component';
-import { ISortOption } from '@app/components/preferenceBar/preferenceBar.interface';
-import { IModalContext, ModalContext } from '@app/components/modals/coreModal/coreModal.context';
+import { IModalContext, ModalContext } from '@shared/interfaces/coreModal.context';
+
+const Main = lazy(() => import('@app/components/main/main.component'));
+const Header = lazy(() => import('@app/components/header/header.component'));
+const Footer = lazy(() => import('@app/components/footer/footer.component'));
```

Comment:
https://reactjs.org/docs/code-splitting.html#reactlazy just take the note into account, we'll be adding SSR in this course(no need to fix right now)